docs/rules/no-reject-literal.md

```diff
@@ -0,0 +1,57 @@
+# disallow using literals as Promise rejection reasons (no-reject-literal)
+
+It is considered good practice to only instances of the built-in `Error` object for user-defined errors in Promises. `Error` objects automatically store a stack trace, which can be used to debug an error by determining where it came from. If a Promise is rejected with a non-`Error` value, it can be difficult to determine where the rejection occurred.
+
+
+## Rule Details
+
+This rule aims to ensure that Promises are only rejected with `Error` objects.
+
+Examples of **incorrect** code for this rule:
+
+```js
+/*eslint no-reject-literal: "error"*/
+
+Promise.reject("something bad happened");
+
+Promise.reject(5);
+
+Promise.reject();
```

Comment:
> There are plenty of use cases where you want to send users down the .catch path without providing an error reason

I'm still not sure I understand what these are. But even if there are valid uses for doing this, it still contradicts the point of this rule, which is to ensure that users who unexpectedly end up on the `.catch` path have useful debugging information rather than something like `Uncaught undefined`.

If someone is intentionally writing code that goes against the point of the rule, even if they have a valid reason for doing so, it seems like it would be better to just turn the rule off (possibly in a file- or line-specific manner). Adding exceptions to the rule would just cause problems for users who *do* want to ensure that Promises are only rejected with errors.